hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3973,24 +3973,36 @@ func reconcileKASNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyp
 		}
 
 		if len(managementClusterNetwork.Spec.ClusterNetwork) > 0 {
-			clusterNetworks := []string{}
-			for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
-				clusterNetworks = append(clusterNetworks, network.CIDR)
-			}
+			var blockedClusterNetworkCIDR string
 
 			// Allow to any destination not on the management cluster service network
 			// i.e. block all inter-namespace egress from KAS not allowed by other rules
-			policy.Spec.Egress = append(policy.Spec.Egress,
-				networkingv1.NetworkPolicyEgressRule{
-					To: []networkingv1.NetworkPolicyPeer{
-						{
-							IPBlock: &networkingv1.IPBlock{
-								CIDR:   "0.0.0.0/0",
-								Except: clusterNetworks,
+			for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
+				ipv4, err := isIPv4(network.CIDR)
+				if err != nil {
+					return fmt.Errorf("error checking the ClusterNetworkCIDR: %v", err)
+				}
+
+				// Set the default
+				if ipv4 {
+					blockedClusterNetworkCIDR = "0.0.0.0/0"
+				} else {
+					blockedClusterNetworkCIDR = "::/0"
+				}
+
+				policy.Spec.Egress = append(policy.Spec.Egress,
+					networkingv1.NetworkPolicyEgressRule{
+						To: []networkingv1.NetworkPolicyPeer{
+							{
+								IPBlock: &networkingv1.IPBlock{
+									CIDR:   blockedClusterNetworkCIDR,
+									Except: []string{network.CIDR},
```

Comment:
Why is this changing the current workflow an putting the whole policy.Spec. within the loop?

Reply:
nvm I see which is to account for both at the same time.

Reply:
could there be more than 2 cluster networks defined? if yes, I think we should first group ipv4 and ipv6 CIRDs then create only 2 egress policies.

Reply:
Technically speaking, it could hold as many as we want but I'm not fully sure if that makes sense for Openshift networking, In my mind it should take 2 as much, to hold both stacks IPv4 and IPv6. I've asked in #forum-sdn to know more about it.

Reply:
No answer from SDN team, so I will asume that there is not a case when more than 2 ClusterNetworks or ServiceNetworks makes sense. I that changes I will create a follow up PR to solve it